finetune/target_models/sequence_labeling.py

```diff
@@ -85,7 +86,47 @@ def _target_encoder(self):
             return SequenceMultiLabelingEncoder(pad_token=self.config.pad_token)
         return SequenceLabelingEncoder(pad_token=self.config.pad_token)
 
-
+def _context_same_cluster(a, b, thresholds=None):
```

Comment:
A potentially cleaner way to organize this function could be to 1. combine the individual `ai`s and `bi`s into one bounding box for a and b; 2. look for X or Y overlap in a and b, 3. look for closeness in the opposite dimension.